app/controllers/users_controller.rb

```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+    
+    def show
+      render json: current_user.as_json(:include => [{:games => {:include => [:sessions]}}, {:sessions => {:include => [:user]}}, {:shared_sessions => {:include => [:user]}}]) || false
```

Comment:
This works, but isn't great to read. I think there are 2 things to consider here:
1. Making this a bit easier to read. We could do something like this:

```suggestion
current_user = User.includes(games: :sessions, sessions: :user, shared_sessions: :user).find(current_user.id)

render json: current_user.as_json(
  include: {
    games: {
      include: :sessions
    },
    sessions: {
      include: :user
    },
    shared_sessions: {
      include: :user
    }
  }
)

      render json: current_user.as_json(:include => [{:games => {:include => [:sessions]}}, {:sessions => {:include => [:user]}}, {:shared_sessions => {:include => [:user]}}]) || false
```
2. With that aside, this isn't very flexible. This makes the assumption that every time you want to fetch the current user, you also always want all of their games, those game sessions, the shared sessions, and so on. With most API's if you sent a request like `GET /user/${userId}`, you would expect to only get the user record back. This API knows too much about what the API wants.

Instead, if you want this additional info you could add `includes` params to your request like this `GET /user/${userId}?include=games,sessions`. This allows the front end to specify what, if any, extra data it needs.

To support this, you would have to modify the controller a bit. Something like this should work (warning: untested code):

```
class UsersController < ApplicationController
  def show
    include_options = build_include_options(params[:include])

    render json: current_user.as_json(include: include_options) || false
  end

  private

  def build_include_options(include_param)
    return {} unless include_param

    include_param.split(',').map do |association|
      case association
      when 'games'
        { games: { include: :sessions } }
      when 'sessions'
        { sessions: { include: :user } }
      when 'shared_sessions'
        { shared_sessions: { include: :user } }
      else
        {}
      end
    end.reduce({}, :merge)
  end
end
```

Reply:
oooh good feedback, thanks! I suspect my other controllers suffer from the same issue, so I'll probably do this in a future PR!